storm/src/main/storm/mesos/schedulers/DefaultScheduler.java

```diff
@@ -54,6 +55,52 @@ public void prepare(Map conf) {
     mesosStormConf = conf;
   }
 
+  private List<MesosWorkerSlot> getMesosWorkerSlots(Map<String, OfferResources> offerResourcesPerNode,
+                                                    Collection<String> nodesWithExistingSupervisors,
+                                                    TopologyDetails topologyDetails) {
+
+    double requestedWorkerCpu = MesosCommon.topologyWorkerCpu(mesosStormConf, topologyDetails);
+    double requestedWorkerMem = MesosCommon.topologyWorkerMem(mesosStormConf, topologyDetails);
+
+    List<MesosWorkerSlot> mesosWorkerSlots = new ArrayList<>();
+    boolean slotFound = true;
+    int slotsNeeded = topologyDetails.getNumWorkers();
+
+    while (slotFound && slotsNeeded > 0) {
```

Comment:
Since `slotFound` isn't actually true yet, this could be misleading.  Let's make this a `do while` instead.